hazelcast-client-new/src/test/java/com/hazelcast/client/cache/stats/ClientCacheStatsTest.java

```diff
@@ -0,0 +1,156 @@
+public class ClientCacheStatsTest extends CacheStatsTestSupport {
+
+    private final TestHazelcastFactory instanceFactory = new TestHazelcastFactory();
+    private HazelcastInstance client;
+
+    @Override
+    protected void onSetup() {
+        super.onSetup();
+        instanceFactory.newHazelcastInstance(createConfig());
+        ClientConfig clientConfig = createClientConfig();
+        client = instanceFactory.newHazelcastClient(clientConfig);
+    }
+
+    @Override
+    protected void onTearDown() {
+        super.onTearDown();
+        instanceFactory.shutdownAll();
+    }
+
+    @Override
+    protected CachingProvider getCachingProvider() {
+        return HazelcastClientCachingProvider.createCachingProvider(client);
+    }
+
+    protected ClientConfig createClientConfig() {
+        return new ClientConfig();
+    }
+
+    @Test
+    public void testStatisticsDisabled() {
+        _testStatisticsDisabled();
+    }
+
+    @Test
+    public void testCreationTime() {
+        _testCreationTime();
+    }
+
+    @Test
+    public void testPutStat() {
+        _testPutStat();
+    }
+
+    @Test
+    public void testAveragePutTimeStat() {
+        _testAveragePutTimeStat();
+    }
+
+    @Test
+    public void testGetStat() {
+        _testGetStat();
+    }
+
+    @Test
+    public void testAverageGetTimeStat() {
+        _testAverageGetTimeStat();
+    }
+
+    @Test
+    public void testRemoveStat() {
+        _testRemoveStat();
+    }
+
+    @Test
+    public void testAverageRemoveTimeStat() {
+        _testAverageRemoveTimeStat();
+    }
+
+    @Test
+    public void testHitStat() {
+        _testHitStat();
+    }
+
+    @Test
+    public void testHitPercentageStat() {
+        _testHitPercentageStat();
+    }
+
+    @Test
+    public void testMissStat() {
+        _testMissStat();
+    }
+
+    @Test
+    public void testMissPercentageStat() {
+        _testMissPercentageStat();
+    }
+
+    @Test
+    public void testLastAccessTimeStat() {
+        _testLastAccessTimeStat();
+    }
+
+    @Test
+    public void testLastUpdateTimeStat() {
+        _testLastUpdateTimeStat();
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testOwnedEntryCount() {
+        ICache<Integer, String> cache = createCache();
+        CacheStatistics stats = cache.getLocalCacheStatistics();
+
+        assertNotNull(stats);
+
+        stats.getOwnedEntryCount();
+    }
+
+    @Test
+    public void testNearCacheStatsWhenNearCacheEnabled() {
+        String cacheName = randomName();
+        CacheConfig cacheConfig = createCacheConfig();
+        cacheConfig.setName(cacheName);
+        ClientConfig clientConfig = ((HazelcastClientProxy) client).getClientConfig();
+        clientConfig.addNearCacheConfig(new NearCacheConfig().setName(cacheName));
+        ICache<Integer, String> cache = createCache(cacheName, cacheConfig);
+        CacheStatistics stats = cache.getLocalCacheStatistics();
+
+        assertNotNull(stats);
```

Comment:
I see in this test and in the tests above that you are testing if statistics is not null. Please create a dedicated test for that and remove this assert from all the other tests. A test that does:

do
assert
do 
assert

Is probably a broken test since it contains multiple situations.